src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -196,13 +199,10 @@ public Stream ReadBlob(string dataTableName, string dataColumnName, long rowId)
             // the one the BLOB handle is open on. Calls to sqlite3_blob_read() and 
             // sqlite3_blob_write() for an expired BLOB handle fail with a return code of
             // SQLITE_ABORT.
-            Stream stream = null;
-            RunInTransaction(() =>
+            return RunInTransaction(((string dataTableName, string dataColumnName, long rowId, SqlConnection connection) tuple) =>
             {
-                stream = ReadBlob_InTransaction(dataTableName, dataColumnName, rowId);
-            });
-
-            return stream;
+                return tuple.connection.ReadBlob_InTransaction(tuple.dataTableName, tuple.dataColumnName, tuple.rowId);
+            }, (dataTableName, dataColumnName, rowId, this));
```

Comment:
instead of a lambda, you might want to make a static-local function, and pass a reference to it.  that makes sure no one accidentally adds a capture in the future (which is pretty easy to do.

Reply:
:+1: Sure, I can do that

Reply:
Sadly, using a local function will not generate a cached `Func<>`, like a lambda would.

Reply:
Ah, got it!

Reply:
wait really?  i thought that was a core value prop of them.  tagging @agocke .

Reply:
@CyrusNajmabadi using: `RunCallback(tuple => localFunction(tuple))` does use a cached lambda, whilst I think `RunCallback(localFunction)` applies method group conversion rules.

Reply:
See https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGA3AWACgUBmLAJgwGEMBvIjRrMgFQCU4oBXAGxgzZcAdlQg8ewCAGMA1gB4WAGgztOvGAD4AFCwwATCDAjKUAVgXLV3PhoxSxE6TICUDJgEgAvLfvjJsrQMjZwJCJgw3RlIMAEshfgARAHstVzCmenTwqIB2AWFRPyctNGUAWQBPQsdZEMjs+vCcADZY+IxK6v8ZLTj+AA9nDG8MftDwgF96+ui+jGSAGQgwYANU+szsphQ8wREHbpLlfuHbToPiwbqshpum7Fa586KAucHT0fGmKcIJoA